libraries/mediaviewer/impl/src/main/kotlin/io/element/android/libraries/mediaviewer/impl/datasource/MediaGalleryDataSource.kt

```diff
@@ -85,11 +86,13 @@ class TimelineMediaGalleryDataSource @Inject constructor(
             }
         }.flatMapLatest {
             timelineMediaItemsFactory.timelineItems
-        }.map { timelineItems ->
-            mediaItemsPostProcessor.process(mediaItems = timelineItems)
-        }.map {
-            mediaTimeline.orCache(it)
-        }.onEach { groupedMediaItems ->
+        }
+        .distinctUntilChanged()
+        .map { timelineItems ->
+            val groupedItems = mediaItemsPostProcessor.process(mediaItems = timelineItems)
+            mediaTimeline.orCache(groupedItems)
+        }
+        .onEach { groupedMediaItems ->
```

Comment:
No actual changes here, but Detekt complained about this so I made some changes to make it happy.